src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -463,20 +467,19 @@ public ResponseEntity<byte[]> getAttachmentUnitAttachmentSlide(@PathVariable Lon
      * @param filename the name of the file
      * @return response entity
      */
-    private ResponseEntity<byte[]> buildFileResponse(String path, String filename) {
+    private ResponseEntity<byte[]> buildFileResponse(Path path, String filename) {
         return buildFileResponse(path, filename, false);
     }
 
     /**
      * Builds the response with headers, body and content type for specified path containing the file name
      *
-     * @param pathString to the file including the file name
-     * @param cache      true if the response should contain a header that allows caching; false otherwise
+     * @param path  to the file including the file name
+     * @param cache true if the response should contain a header that allows caching; false otherwise
      * @return response entity
      */
-    private ResponseEntity<byte[]> buildFileResponse(String pathString, boolean cache) {
-        Path path = Path.of(pathString);
-        return buildFileResponse(path.getParent().toString(), path.getFileName().toString(), cache);
+    private ResponseEntity<byte[]> buildFileResponse(Path path, boolean cache) {
+        return buildFileResponse(path.getParent(), path.getFileName().toString(), cache);
```

Comment:
Isn't `.toString()` missing with `path.getParent()`?

Reply:
No, The first parameter is now a path, not a string.